pkg/compact/compact.go

```diff
@@ -879,10 +879,13 @@ func NewBucketCompactor(
 }
 
 // Compact runs compaction over bucket.
-func (c *BucketCompactor) Compact(ctx context.Context) error {
+func (c *BucketCompactor) Compact(ctx context.Context) (err error) {
```

Comment:
I think the whole diff would have been easier if you just named the returned error differently, like `returnErr`.